saleor/order/migrations/tasks/saleor3_9.py

```diff
@@ -0,0 +1,131 @@
+from decimal import Decimal
+from celery.utils.log import get_task_logger
+from django.db.models import Q, F
+
+
+from ....tax.models import TaxConfiguration
+from ....channel.models import Channel
+from ....order.models import OrderLine, Order
+from ....celeryconf import app
+
+task_logger = get_task_logger(__name__)
+
+BATCH_SIZE = 500
+
+
+def add_taxes_to_price(price, tax_rate, prices_entered_with_tax):
+    tax_rate = Decimal(1 + tax_rate)
+
+    if prices_entered_with_tax:
+        net_amount = price / tax_rate
+        gross_amount = price
+    else:
+        net_amount = price
+        gross_amount = price * tax_rate
+
+    return net_amount, gross_amount
+
+
+def update_fields_for_line(
+    field_name, line, tax_rate, prices_entered_with_tax, lines_to_update
+):
+    line_net = getattr(line, f"{field_name}_net_amount")
+    line_gross = getattr(line, f"{field_name}_gross_amount")
+    if line_net == line_gross and all([line_net != 0, line_gross != 0]):
+        line_net, line_gross = add_taxes_to_price(
+            line_net, tax_rate, prices_entered_with_tax
+        )
+        setattr(line, f"{field_name}_net_amount", line_net)
+        setattr(line, f"{field_name}_gross_amount", line_gross)
```

Comment:
Shouldn't we quantize price value before saving?

Reply:
yes. fixed